stored_requests/fetcher.go

```diff
@@ -52,16 +52,12 @@ type fetcherWithCache struct {
 
 func (f *fetcherWithCache) FetchRequests(ctx context.Context, ids []string) (data map[string]json.RawMessage, errs []error) {
 	data = f.cache.GetRequests(ctx, ids)
-	if len(data) >= len(ids) {
-		return data, nil
-	}
 
-	leftoverIds := make([]string, len(ids)-len(data))
-	currentIndex := 0
+	// Fixes #311
+	leftoverIds := make([]string, 0, len(ids)-len(data))
```

Comment:
What is this leftoverIds used for? what are ids and what is data?

Reply:
`ids` are the IDs of the Stored Request which someone wants to fetch.
`data` is a map from ID to Stored Request. This contains everything the cache returns

`leftoverIds` is the subset of `ids` which _aren't_ included in `data`. After that loop, it should only have the elements of `ids` which were _not_ found in the Cache.

Reply:
I don't see differences from this diff. Both are sending the same set of leftoverids to Fetcher later, right?

Reply:
They're not. If `ids` has duplicate values, then the `map` will have one fewer, because maps automatically filter out duplicates... so the lengths were wrong.

If you run the regression test I added against the `master` code, you can see it fail.